common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -225,76 +246,137 @@ public String encrypt(final String clearText)
 
         final String encryptedText = new String(Base64.encode(blobVerAndEncryptedDataAndIVAndMacDigest,
                 Base64.NO_WRAP), AuthenticationConstants.ENCODING_UTF8);
-        Log.v(TAG, "Finished encryption");
+        Logger.verbose(TAG + methodName, "Finished encryption");
 
         return getEncodeVersionLengthPrefix() + ENCODE_VERSION + encryptedText;
     }
 
     @Override
     public String decrypt(final String encryptedBlob) throws GeneralSecurityException, IOException {
-        final String methodName = "decrypt";
-        Log.v(TAG, "Starting decryption");
+        final String methodName = ":decrypt";
+        Logger.verbose(TAG + methodName, "Starting decryption");
 
         if (StringExtensions.isNullOrBlank(encryptedBlob)) {
             throw new IllegalArgumentException("Input is empty or null");
         }
 
-        int encodeVersionLength = encryptedBlob.charAt(0) - 'a';
-        validateEncodeVersion(encryptedBlob, encodeVersionLength);
+        if (getEncryptionType(encryptedBlob) == EncryptionType.UNENCRYPTED) {
+            Logger.verbose(TAG + methodName, "This string is not encrypted. Finished decryption.");
+            return encryptedBlob;
+        }
 
-        final byte[] bytes = Base64.decode(
-                encryptedBlob.substring(1 + encodeVersionLength),
-                Base64.DEFAULT
-        );
+        final String packageName = getPackageName();
+        final List<KeyType> potentialDecryptionKeyTypeList = initializeDecryptionKeyTypeList(encryptedBlob, packageName);
```

Comment:
getKeysForEncryptionType?

Reply:
Why not just return the keys here... rather than the list of key types?

Reply:
After scoping down the work, this is no longer needed. (This is to prevent making unnecessary IPC calls in an edge case - we only invoke them right before we're decrypting - not when we're initializing the list).

I'll just return the list of keys as suggested, as that's more straightforward.